cloud/passportauth/src/auth/PassportSetup.ts

```diff
@@ -0,0 +1,73 @@
+import * as express from 'express';
+import * as session from 'express-session';
+import * as passport from 'passport';
+import {Strategy} from 'passport-local';
+import BaseUser from '../user/BaseUser';
+import UserSecService from '../user/UserSec';
+
+export interface Setup {
+  init(app: express.Express): void;
+  setStrategy(): void;
+  setSerializeUser(): void;
+  setDeserializeUser(): void;
+}
+
+class PassportSetup<T extends BaseUser> {
+  protected userSec: UserSecService<T>;
+
+  constructor(protected readonly userSecService: UserSecService<T>) {
+    this.userSec = userSecService;
+  }
+
+  public init(app: express.Express) {
+    app.use(session({secret: 'test', resave: false, saveUninitialized: false}));
+    app.use(passport.initialize());
+    app.use(passport.session());
+
+    this.setStrategy();
+    this.setSerializeUser();
+    this.setDeserializeUser();
+  }
+
+  public setStrategy() {
+    passport.use(new Strategy((username, password, done) => {
```

Comment:
This is the passport local strategy. Similar to this example here: https://github.com/passport/express-4.x-local-example/blob/master/server.js